pkg/compose/start.go

```diff
@@ -28,15 +28,22 @@ import (
 	"github.com/docker/compose/v2/pkg/progress"
 )
 
-func (s *composeService) Start(ctx context.Context, project *types.Project, options api.StartOptions) error {
+func (s *composeService) Start(ctx context.Context, projectName string, options api.StartOptions) error {
 	return progress.Run(ctx, func(ctx context.Context) error {
-		return s.start(ctx, project, options, nil)
+		return s.start(ctx, projectName, options, nil)
 	})
 }
 
-func (s *composeService) start(ctx context.Context, project *types.Project, options api.StartOptions, listener api.ContainerEventListener) error {
-	if len(options.AttachTo) == 0 {
-		options.AttachTo = project.ServiceNames()
+func (s *composeService) start(ctx context.Context, projectName string, options api.StartOptions, listener api.ContainerEventListener) error {
+	var containers Containers
+	containers, err := s.getContainers(ctx, projectName, oneOffExclude, true, options.AttachTo...)
```

Comment:
Containers are used in the `projectFromName` function to detect and populate the services section of the project. 
If we pass the `options.AttachTo` array of services to `getContainers`, we'll only get the containers running services declared in the command line. 

If your compose stack declare services with dependencies you could get `no such service: XXX` error.
For example if you try the sample `nginx-golang-mysql` from the `awesome-compose` repository and try the following command `docker compose --project-name nginx-golang-mysql start proxy` you will get this error
`no such service: backend`

So we need to get all the all containers from a project and not only the ones linked to services passed as parameter of the command

```suggestion
	containers, err := s.getContainers(ctx, projectName, oneOffExclude, true)
```

Reply:
Thanks for pointing this out
I made the change and tested it with `nginx-golang-mysql` and it appears to be fixed.